helper/write.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * Fedora Media Writer
+ * Copyright (C) 2016 Martin Bříza <mbriza@redhat.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#include "write.h"
+
+#include <unistd.h>
+
+#include <QTextStream>
+#include <QtGlobal>
+
+int onProgress(void *data, long long offset, long long total) {
+    constexpr long long MAGIC = 234;
+    ProgressStats &stats = *static_cast<ProgressStats *>(data);
```

Comment:
Why use reference instead of the pointer?

Reply:
Because
* C++
* I prefer the `.` operator.
* I isn't any overhead even if I wouldn't need to cast it and store it somewhere anyways.
* References are safer to use since it's more unlikely that I accidentally put them in a state that causes crashes.

Reply:
>* C++
Pointers are C++ as well.
>*I prefer the . operator.
I prefer not having lvalue references in my code.
>*I isn't any overhead even if I wouldn't need to cast it and store it somewhere anyways.
Yeah, I think it boils down to the same binary code in the end.
>*References are safer to use since it's more unlikely that I accidentally put them in a state that causes crashes.
Doesn't apply to this case though. This is just a weird alias for a pointer.
This is not a bug or anything near that. I just think it's good to call the things the right name and this is a pointer. Dereferecing it and storing it into a local reference doesn't change the fact you'll be writing to the same piece of memory using the same pointer you have been passed from the calling method.